tests/unit/components/mapSettingsMenu/countrySelector.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { render } from "@testing-library/vue";
+import { describe, expect, test } from "vitest";
+import { mockMapSettings, mockPinia } from "../../mocks/mockPinia";
+import CountrySelector from "../../../../src/components/mapSettingsMenu/CountrySelector.vue";
+import { mockRouter } from "../../mocks/mockRouter";
+import { mockVuetify } from "../../mocks/mockVuetify";
+
+const router = mockRouter();
+
+const renderComponent = (country) => {
+    const store = mockPinia({
+        mapSettings: mockMapSettings({ country })
+    });
+    return render(CountrySelector, {
+        global: {
+            plugins: [store, mockVuetify, router]
+        }
+    });
+};
+
+describe("CountrySelector", () => {
+    test("renders as expected on the global view", async () => {
+        const { findByText } = renderComponent("");
+        expect(await findByText("Global")).toBeVisible();
+    });
+
+    test("renders as expected on country view", async () => {
+        const { findByText } = renderComponent("MWI");
+        expect(await findByText("Malawi")).toBeVisible();
+    });
+});
```

Comment:
Is this the sort of thing you were trying to do here?
```
test("items appear when click input", async () => {
        await renderComponent();
        const user = userEvent.setup();
        const input = await screen.findByPlaceholderText("Start typing to filter");
        await user.click(input);
        const globalListItem = await screen.findByText("Global");
        expect(globalListItem).toBeVisible();
        expect((globalListItem as HTMLElement).className).toContain("v-list-item-title");
        const mwiListItem = await screen.findByText("Malawi");
        expect(mwiListItem).toBeVisible();
        expect((mwiListItem as HTMLElement).className).toContain("v-list-item-title");
    });
```

Reply:
Yes, but with either typing into the input or clicking the dropdown, neither of which I could make work in the unit test.

Reply:
I'll include this test that doesn't do filtering, to avoid the perfect being the enemy of the good.

Reply:
How about this? 
```
const input = await screen.findByPlaceholderText("Start typing to find a country");
await user.type(input, "Tan");
await waitFor(() => expect(screen.queryByText("Malawi")).toBe(null));
```

Reply:
The issue is in locating the dropdown, I can't seem to find it / see it (it probably is failing to exist for some reason)

Reply:
I'd sort of assumed that if you could see the Malawi list item:
```
const mwiListItem = await screen.findByText("Malawi");
expect(mwiListItem).toBeVisible();
```
then that must mean that the drop down is open and showing its items.. 
Are you wanting to check for the existence of an element with a particular class name, and was that failing?

Reply:
I was literally seeing no text that I expected from the dropdown, when using screen.debug(). But now we have something that works.